analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/PubliclyWritableDirectoriesBase.cs

```diff
@@ -0,0 +1,110 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Resources;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class PubliclyWritableDirectoriesBase<TSyntaxKind, TInvocationExpression> : HotspotDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TInvocationExpression : SyntaxNode
+    {
+        protected const string DiagnosticId = "S5443";
+        private const string MessageFormat = "Make sure publicly writable directories are used safely here.";
+        private const RegexOptions WindowsAndUnixOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant;
+        private const string VariableSlashTwoPartTemplate = @"^[\\\/]{0}[\\\/]{1}([\\\/]|$)";
+        private const string ForwardSlashTwoPartTemplate = @"^\/{0}\/{1}(\/|$)";
+        private const string ForwardSlashThreePartTemplate = @"^\/{0}\/{1}\/{2}(\/|$)";
+        private const string Temp = @"^([a-z]:[\\\/]?|[\\\/][\\\/][^\\\/]+[\\\/]|[\\\/])(windows[\\\/])?te?mp([\/\\]|$)";
+        private readonly Regex userProfile = new Regex(@"^%USERPROFILE%[\\\/]AppData[\\\/]Local[\\\/]Temp([\\\/]|$)", WindowsAndUnixOptions);
+        private readonly string varTmp = string.Format(VariableSlashTwoPartTemplate, "var", "tmp");
+        private readonly string usrTmp = string.Format(VariableSlashTwoPartTemplate, "usr", "tmp");
+        private readonly string devShm = string.Format(VariableSlashTwoPartTemplate, "dev", "shm");
+        private readonly string devMqueue = string.Format(ForwardSlashTwoPartTemplate, "dev", "mqueue");
+        private readonly string runLock = string.Format(ForwardSlashTwoPartTemplate, "run", "lock");
+        private readonly string varRunLock = string.Format(ForwardSlashThreePartTemplate, "var", "run", "lock");
+        private readonly string libraryCaches = string.Format(ForwardSlashTwoPartTemplate, "library", "caches");
+        private readonly string usersShared = string.Format(ForwardSlashTwoPartTemplate, "users", "shared");
+        private readonly string privateTmp = string.Format(ForwardSlashTwoPartTemplate, "private", "tmp");
+        private readonly string privateVarTmp = string.Format(ForwardSlashThreePartTemplate, "private", "var", "tmp");
+        private readonly DiagnosticDescriptor rule;
+        private readonly Regex linuxPubliclyWritabelDirs;
```

Comment:
To keep things simple and aligned with my other comment. We're inside `PubliclyWritableDirectoriesBase` scope.
```suggestion
        private readonly Regex linuxDirs;
```